omod/src/main/webapp/resources/scripts/conditionlist/controllers/editcondition.controller.js

```diff
@@ -0,0 +1,145 @@
+                window.location = '/' + OPENMRS_CONTEXT_PATH + '/coreapps/conditionlist/manageConditions.page?patientId=' + $scope.patientUuid + '&';
+            }, function (error) {
+                //emr.errorAlert("conditionlist.updateCondition.error");
+                emr.errorAlert("Error Saving condition");
+            });
+        }
+    }
+
+    self.initCondition = self.initCondition || function () {
+            $scope.patientUuid = CommonFunctions.extractUrlArgs(window.location.search)['patientId'];
+            $scope.condition = new ConditionModel($scope.patientUuid)
+            $scope.condition = CommonFunctions.extractUrlArgs(window.location.search)['condition'];
+            $scope.condition = JSON.parse($scope.condition.split("%22").join("\"").split("%20").join(" "));
+
+            $scope.concept = {
+                display: $scope.condition.concept.name,
+                concept: {
+                    display: $scope.condition.concept.name,
+                },
+                conceptName: {
+                    display: $scope.condition.concept.name,
+                }
+            }
+
+            var onsetDatePicker = angular.element(document.getElementsByName('conditionStartDate'))[0];     
+            onsetDatePicker.value = self.convertDate1($scope.condition.onSetDate);
+          
+            if($scope.condition.status=='INACTIVE'){
+                var endDatePicker = angular.element(document.getElementsByName('conditionEndDate'))[0];      
+                endDatePicker.value = self.convertDate1($scope.condition.endDate);
+            } 
+
+            var inputs = document.getElementsByTagName('input');
+            inputs[1].value = self.convertDate2($scope.condition.onSetDate);  
+            self.showEndDate();
+        }
+
+    self.showEndDate = self.showEndDate || function() {
+        var inputs = document.getElementsByTagName('input');
+        var groups = document.getElementsByClassName("group");
+        if($scope.condition.status=='INACTIVE'){
+            groups[2].style.visibility = "visible";
+            inputs[3].value = self.convertDate2($scope.condition.endDate);
+        }
+        else{
+            groups[2].style.visibility = "hidden";    
+        }
+    }
+
+    self.validateCondition = self.validateCondition || function () {
+            var concept ;
+            if($scope.concept.concept.uuid)
+            {
+                concept=$scope.concept.concept;
+            }
+            else{
+                concept=$scope.condition.concept;
+            }
+            if (concept !== null) {
+                $scope.condition.concept = new ConceptModel(concept.uuid, concept.display);
+            } else {
+                // noncoded diagnosis
+                $scope.condition.conditionNonCoded = "NON_CODED:" + $scope.concept.word;
+            }
+
+            $scope.condition.onSetDate = self.getSelectedDate();
+            if($scope.condition.status=='INACTIVE')
```

Comment:
Shouldn't you have a constant for this?

Reply:
I will update the code

Reply:
Is it possible to call the constants defined in properties file in JS? or do i create a global variable for this in the JS file itself?